client/components/CartView.js

```diff
@@ -1,98 +1,86 @@
-import React, { useEffect, useState } from 'react';
-import { clearCart, getCart, removeItemFromCart, buyCart } from '../store/cart';
-import { useDispatch, useSelector } from 'react-redux';
-import Container from 'react-bootstrap/Container';
-import Button from 'react-bootstrap/Button';
-import Table from 'react-bootstrap/Table';
-import Card from 'react-bootstrap/Card';
-import ButtonGroup from 'react-bootstrap/ButtonGroup';
-import Row from 'react-bootstrap/Row';
-import Col from 'react-bootstrap/Col';
-import Image from 'react-bootstrap/Image';
-import history from '../history';
+import React, { useEffect, useState } from "react";
+import { clearCart, getCart, removeItemFromCart, buyCart } from "../store/cart";
+import { useDispatch, useSelector } from "react-redux";
+import { editCart } from "../store/cart";
+import Container from "react-bootstrap/Container";
+import Button from "react-bootstrap/Button";
+import Table from "react-bootstrap/Table";
+import Card from "react-bootstrap/Card";
+import ButtonGroup from "react-bootstrap/ButtonGroup";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import Image from "react-bootstrap/Image";
+import history from "../history";
 
 const CartView = () => {
-  const [editMode, toggleEditMode] = useState(false);
+  const dispatch = useDispatch();
+  const [emailInput, setEmailInput] = useState("");
+  const [guestEmail, setGuestEmail] = useState("");
 
   const { id, username } = useSelector((state) => state.auth);
-  //grab user ID and username from state
+  const cart = useSelector((state) => state.cart);
 
-  const dispatch = useDispatch();
   useEffect(() => {
-    if (id) {
-      dispatch(getCart(id));
-    }
+    dispatch(getCart(id));
   }, [id]);
-  // when the page loads (and whenever user login changes), dispatch getCart
-  const cart = useSelector((state) => state.cart);
-  // put that cart in local scope
 
-  function handleRemove(cartItem, userId = 0) {
-    // removes individual items from cart
+  function handleRemoveItem(cartItem, userId = 0) {
     dispatch(removeItemFromCart(cartItem, userId));
   }
 
+  const handleEdit = (e, cartItem, userId = 0) => {
```

Comment:
remember to remove the ```userId = 0``` wherever handlers are dispatching stuff with a userId, so that your ```!userId``` checks in the thunks work correctly